src/python/espressomd/system.pyx

```diff
@@ -442,3 +443,59 @@ cdef class System(object):
             """
             auto_exclusions(distance)
 
+
+    def _is_valid_type(self, current_type):
+        return (not (isinstance(current_type, int) or current_type < 0 or current_type > globals.n_particle_types))
+
+
+    def check_valid_type(self, current_type):
+        if self._is_valid_type(current_type):
+            raise ValueError("type", current_type, "does not exist!")
+
+
+    def setup(self, type_list=None):
+        """
+        For using Espresso conveniently for simulations in the grand canonical
+        ensemble, or other purposes, when particles of certain types are created
+        and deleted frequently. Particle ids can be stored in lists for each
+        individual type and so random ids of particles of a certain type can be
+        drawn. If you want Espresso to keep track of particle ids of a certain type
+        you have to initialize the method by calling the setup function. After that
+        Espresso will keep track of particle ids of that type.
+
+        """
+        if not hasattr(type_list, "__iter__"):
+            raise ValueError("type_list has to be iterable.")
+
+        for current_type in type_list:
+            if (max_seen_particle < 0):
```

Comment:
Your core code can perfectly handle types w/o parts, so this check is not needed.